README.md

```diff
@@ -1,66 +1,124 @@
- - Typed Rust structure from any plain JavaScript object (`{ key1: value1, ... }`).
- - Rust enum from either a string (`"Variant"`) or a plain object. Specific representation is [controlled](https://serde.rs/enum-representations.html) by `#[serde(...)]` attributes and should be compatible with `serde-json`.
-
-Serialization is compatible with the deserialization, but it's limited to a single representation, so it chooses:
- - `undefined` for `()` or `None`.
- - ES2015 `Map` for Rust maps (can be configured to use plain objects via `serialize_maps_as_objects(true)`).
- - `Array` for any Rust sequences.
- - `Uint8Array` for byte buffers.
- - Plain JavaScript object for typed Rust structures.
+Likewise, the `from_value` function can be used for deserialization.
+
+## Supported Types
+
+Note that, even though it might often be the case, this library doesn't attempt
+to be strictly compatible with either [`serde_json`][serde_json] or,
+correspondingly, `JsValue::from_serde` / `JsValue::into_serde`, instead
+prioritising better compatibility with common JavaScript idioms and
+representations.
+
+[serde_json]: https://docs.serde.rs/serde_json/
+
+### Deserialization
+
+| Javascript Type                            | Rust Type                          |
+| ------------------------------------------ | ---------------------------------- |
+| `undefined` and `null`                     | `()` or `Option<T>`                |
+| `false` and `true`                         | `bool`                             |
+| Any [safe integer]                         | `u8`/`i8`/.../`u128`/`i128`        |
+| JS Number                                  | `f32` or `f64`                     |
+| Length-1 String                            | `char`                             |
+| String                                     | `String` or `Cow<'static, str>`    |
+| `"Variant"`                                | Enum Variant<sup>†</sup>           |
+| `{ key1: value1, ... }` Object             | `HashMap<String, T>` or Struct `T` |
+| Any Iterable of `[key, value]`<sup>‡</sup> | `HashMap`, `BTreeMap`, etc.        |
+| Any Iterable of `value`                    | Tuple, `Vec`, `HashSet`, etc.      |
+| `ArrayBuffer` or `Uint8Array`              | [serde_bytes] Byte Buffer          |
+
+**Notes:**
+
+- †: The specific representation is [controlled] by `#[serde(...)]` attributes
+  and should be compatible with `serde-json`.
+- ‡: Excepts are [internally tagged] and [untagged] enums. These representations
+  currently do not support deserializing map-like iterables. They only support
+  deserialization from `Object` due to their special treatment in `serde`. This
+  restriction may be lifted at some point in the future if a `serde(with = ...)`
+  attribute can define the expected Javascript representation of the variant, or
+  if [serde-rs/serde#1183] gets resolved.
+
+[safe integer]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger
```

Comment:
Personally I prefer to keep links inline as it makes it easier to edit/remove them without looking for definitions somewhere else in the file.

Reply:
I always keep the links underneath the paragraph they belong to, meaning they're always in one place and thus easy to find, but if you prefer inlined I can revert.

Reply:
Yeah I rarely want links together with each other, but rather with their surrounding contexts, so I'd prefer to revert.

Reply:
I've inlined all the URLs except two; the ones that appear in the chart above. Otherwise the chart would get quite ugly.